### ISSUE TITLE
Style form elements differently when not in insert mode?

### ISSUE THREAD

**Issue by mgraham** [The-Compiler/qutebrowser#581]
When focus moves to a form field, I find I tend to look at the field, not the status line. Would it be possible to style form fields differently when they are focused and in insert mode?  Or even change the cursor style?

In gvim, I've come to rely on the fact that the cursor changes from a block (normal mode) to a vertical bar (insert mode).

**Comment by The-Compiler** [comment]
Ideally, qutebrowser would always be in insert mode when a form field is focused - however sometimes when one gets auto-focused or is some special kind of input field, that doesn't work properly (see #89, #253 ).

Changing the cursor shape would indeed be a nice solution for that - I'm not sure whether it's possible or not, but I'll take a look at it when I get the time.

**Comment by mgraham** [comment]
Great - thanks for looking into it!

**Comment by steveb-123** [comment]
I'm curious if this is still on the todo list? 

I've been a user for years and this feels like the Last Great Problem of qutebrowser.

I wonder if I am alone in accidentally hammering 20 keystrokes of commands whenever I see a blinking cursor but insert is not triggered.. It might not be such a problem for purely keyboard-drivers. For me, it frequently causes me to lose part-written emails or mess up forms etc.

I have often wondered if detecting the visual blinking cursor might be programmatically possible, but I have no idea how those are rendered as I know nothing about html, css etc presently. For me, the blinking cursor symbol is the cue to start typing, so if insert mode could either detect this, or block this I think it would prevent my errors... and the inevitable ensuing micro-rage :D

**Comment by The-Compiler** [comment]
> I'm curious if this is still on the todo list?

Yes, hence why this issue is still open. But a lot of things are.

> I've been a user for years and this feels like the Last Great Problem of qutebrowser.

Consider that there are many more users than developers, and every one of those probably has a different issue which is most important to them.

> I have often wondered if detecting the visual blinking cursor might be programmatically possible

It's a lot trickier than you'd expect. #2471 has some ideas, and some parts of #5376 might actually tie into it too.

**Comment by steveb-123** [comment]
>Consider that there are many more users than developers, and every one of those probably has a different issue which is most important to them.

I apologise for the bump, qutebrowser has been basically flawless for me for the last couple of years now so it's been easy to forget this. I had assumed that this issue must be affecting everyone, every day* and it seemed curious that it was 5 years old. If I'd spotted #2471 I certainly wouldn't have brought it up. 

Apologies again and thanks for all the work on qutebrowser.

*As I think about this... Why does this seem like such a glaring issue to me but not to others/everyone/you... I wonder if it may be a problem for people who are not regularly also using vim through a terminal emulator. Dynamically switching the caret style -from rectangular to thin- really is the main visual confirmation of input-mode for me, rather than looking to the modeline in the editor. As I understand it dynamically switching caret-shape within a session is not default vim behaviour in terminals. Perhaps the terminal vimmers are used to subconsiously checking the modeline in a way people using gvim, evil-emacs/spacemacs/doom do not do... and thus this problem doesn't arise for them.
